app/Models/Product.php

```diff
@@ -34,12 +37,229 @@ class Product extends Model
+                $image = $this->getFirstMedia('product_images', function ($media) { return $media->name === 'secondary_image'; });
+
+                return $image ? $image->getUrl() : null;
+            },
+            set: function ($value) {
+                $previousImage = $this->getFirstMedia('product_images', function ($media) {
+                    return $media->name === 'secondary_image';
+                });
+
+                if ($value) {
+                    if ($previousImage && \Str::startsWith($value, 'data:image')) {
+                        $previousImage->delete();
+                    }
+                    if (\Str::startsWith($value, 'data:image')) {
+                        $extension = '.'.\Str::after(mime_content_type($value), '/');
+                        $this
+                            ->addMediaFromBase64($value)
+                            ->usingName('secondary_image')
+                            ->usingFileName('secondary_image'.$extension)
+                            ->toMediaCollection('product_images', 'products');
+                    }
+
+                    return;
+                }
+
+                if ($previousImage) {
+                    $previousImage->delete();
+                }
+            }
+        );
+    }
+
+    public function mainImage(): \Illuminate\Database\Eloquent\Casts\Attribute
+    {
+        return \Illuminate\Database\Eloquent\Casts\Attribute::make(
+            get: function () {
+                $image = $this->getFirstMedia('product_images', function ($media) { return $media->name === 'main_image'; });
+
+                return $image ? $image->getUrl() : null;
+            },
+            set: function ($value) {
+                $previousImage = $this->getFirstMedia('product_images', function ($media) {
+                    return $media->name === 'main_image';
+                });
+
+                if ($value) {
+                    if ($previousImage && \Str::startsWith($value, 'data:image')) {
+                        $previousImage->delete();
+                    }
+                    if (\Str::startsWith($value, 'data:image')) {
+                        $extension = '.'.\Str::after(mime_content_type($value), '/');
+                        $uploadedMedia = $this
+                            ->addMediaFromBase64($value)
+                            ->usingName('main_image')
+                            ->usingFileName('main_image'.$extension)
+                            ->toMediaCollection('product_images', 'products');
+                    }
+
+                    return $uploadedMedia->id ?? $previousImage->id;
+                }
+
+                if ($previousImage) {
+                    $previousImage->delete();
+                }
+            }
+        );
+    }
+
+    public function specifications(): \Illuminate\Database\Eloquent\Casts\Attribute
+    {
+        return \Illuminate\Database\Eloquent\Casts\Attribute::make(
```

Comment:
Let's do this as events in the CrudController, in Backpack itself.

```php
CRUD::field('specifications')->type('upload_multiple')->events([
    'saving' => function ($entry) {
        $entry->uploadMediaToCollection($media_name, $collection_name, $disk, $filename);
    },
    'retrieved' => function ($entry) {
        $entry->specifications = $entry->getMediaFromCollection($media_name, $collection_name, $disk);
    },
]);
```

Reply:
Note: we'd still need 3 methods (not really with these names):
    => uploadImageToCollection()
    => uploadMediaToCollection()
    => uploadMultipleMediaToCollection()

They can live in CrudTrait, no problem. We just don't call them in the Model accessor/mutator, to not take over the Model.

Reply:
If we move this into a `backpack/spatie-media-library-events` package we would:
- PRO - be able to tie that package to `spatie/media-library` package
- PRO - see download stats
- CON - have to maintain an extra micro-package;